#! /bin/sh /usr/share/dpatch/dpatch-run
## 39_check_ircd_fix_epn.dpatch by Alexander Wirt <formorer@formorer.de>
## From: Holger Weiss <holger@zedat.fu-berlin.de>
## Date: Mon, 12 Apr 2010 13:36:33 +0000 (+0200)
## Subject: Fix Failure when run via ePN
## X-Git-Url: http://nagiosplug.git.sf.net/git/gitweb.cgi?p=nagiosplug%2Fnagiosplug;a=commitdiff_plain;h=7a661c7aed2a0bad4ce1e3786bc707cd2b07c128
##
## DP: Fixes epn in check_ircd (#545940) (bugs.debian.org/545940)
## DP: Upstream bug is: https://sourceforge.net/tracker/?func=detail&atid=397597&aid=2864973&group_id=29880

@DPATCH@

diff --git a/plugins-scripts/check_ircd.pl b/plugins-scripts/check_ircd.pl
index 3fbce2e..42a9bca 100755
--- a/plugins-scripts/check_ircd.pl
+++ b/plugins-scripts/check_ircd.pl
@@ -187,19 +187,19 @@ MAIN:
 
 	if ($opt_h) {print_help(); exit $ERRORS{'OK'};}
 
-	($opt_H) || ($opt_H = shift) || usage("Host name/address not specified\n");
+	($opt_H) || ($opt_H = shift @ARGV) || usage("Host name/address not specified\n");
 	my $remotehost = $1 if ($opt_H =~ /([-.A-Za-z0-9]+)/);
 	($remotehost) || usage("Invalid host: $opt_H\n");
 
-	($opt_w) || ($opt_w = shift) || ($opt_w = 50);
+	($opt_w) || ($opt_w = shift @ARGV) || ($opt_w = 50);
 	my $warn = $1 if ($opt_w =~ /^([0-9]+)$/);
 	($warn) || usage("Invalid warning threshold: $opt_w\n");
 
-	($opt_c) || ($opt_c = shift) || ($opt_c = 100);
+	($opt_c) || ($opt_c = shift @ARGV) || ($opt_c = 100);
 	my $crit = $1 if ($opt_c =~ /^([0-9]+)$/);
 	($crit) || usage("Invalid critical threshold: $opt_c\n");
 
-	($opt_p) || ($opt_p = shift) || ($opt_p = 6667);
+	($opt_p) || ($opt_p = shift @ARGV) || ($opt_p = 6667);
 	my $remoteport = $1 if ($opt_p =~ /^([0-9]+)$/);
 	($remoteport) || usage("Invalid port: $opt_p\n");
 
